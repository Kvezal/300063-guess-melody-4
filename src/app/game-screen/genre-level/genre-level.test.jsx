import React from "react";
import {MemoryRouter} from "react-router-dom";
import render from "react-test-renderer";

import GenreLevel from "./genre-level";


describe(`Genre`, () => {
  const question = {
    type: `genre`,
    genre: `rock`,
    answers: [
      {src: `src/1`, genre: `rock`},
      {src: `src/2`, genre: `blues`},
      {src: `src/3`, genre: `jazz`},
      {src: `src/4`, genre: `rock`}
    ],
  };

  test(`should render component`, () => {
    const tree = render
      .create(
          <MemoryRouter initialEntries={[`/game`]}>
            <GenreLevel
              question={question}
              onAnswer={() => {}}
              onAnswerChange={() => {}}
              answers={question.answers.map(() => false)}
              renderPlayer={() => {}}
            />
          </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

