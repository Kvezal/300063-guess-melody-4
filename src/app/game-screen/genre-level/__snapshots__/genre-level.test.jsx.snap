// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre should render component 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите инди-рок треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <button
        className="play-button play-button--small "
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="src/1"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="rock"
        />
        <label
          className="game__check"
          htmlFor="src/1"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="play-button play-button--small "
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="src/2"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="blues"
        />
        <label
          className="game__check"
          htmlFor="src/2"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="play-button play-button--small "
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="src/3"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="jazz"
        />
        <label
          className="game__check"
          htmlFor="src/3"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="play-button play-button--small "
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="src/4"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="rock"
        />
        <label
          className="game__check"
          htmlFor="src/4"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
